src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
No-one external to the object store has the ability to modify the `SyncConfig` that `Realm` and `SyncSession` instances hold, so that's not a concern. I'm asking if those `SyncConfig` instances that the object store holds internally outlive the objects that use the `ChangesetCooker`. If they don't, the object may be deallocated while it's still being used.